pyvista/plotting/renderer.py

```diff
@@ -1868,6 +1868,7 @@ def clear(self):
                 except KeyError:
                     pass
 
+        self._charts.deep_clean()
```

Comment:
Accessing `self._charts` in `clear()` (even without the `deep_clean()`) leads to GC errors in `test_plotting.py`, due to some `dataset` `vtkWeakReference` not being collecting. Trying to figure this out...

Reply:
I guess the issue was related to creating a new `Charts` instance via the `self._charts` property when it was `None` earlier. Adding a less elegant explicit check for `self.__charts is not None` seems to make the GC errors go away, while still fixing the original issue. I'm open to alternate suggestions though.